### ISSUE TITLE
Update `tf.stack` for numpy casting 

### ISSUE THREAD

**Issue by brendancol** [bokeh/datashader#255]
numpy ufunc `over` is causing trouble because of update to numpy which doesn't allow unsafe casting from float -> int.  Looking like this is a the toolz level

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-14-8c2b2b42392b> in <module>()
     11 
     12 p = base_plot(x_range=(-11020645, -10797986), y_range=(3503546, 3632767))
---> 13 InteractiveImage(p, hillshade_raster)

/Users/bcollins/anaconda/envs/training/lib/python3.5/site-packages/datashader/bokeh_ext.py in __init__(self, bokeh_plot, callback, delay, timeout, throttle, **kwargs)
    130 
    131         # Initialize the image and callback
--> 132         self.ds, self.renderer = self._init_image()
    133         callback = self._init_callback()
    134         self.p.x_range.callback = callback

/Users/bcollins/anaconda/envs/training/lib/python3.5/site-packages/datashader/bokeh_ext.py in _init_image(self)
    173         y_range = (ymin, ymax)
    174         dw, dh = xmax - xmin, ymax - ymin
--> 175         image = self.callback(x_range, y_range, width, height, **self.kwargs)
    176 
    177         ds = ColumnDataSource(data=dict(image=[image.data], x=[xmin],

<ipython-input-14-8c2b2b42392b> in hillshade_raster(x_range, y_range, w, h, how)
      8     agg = cvs.raster(raster_data)
      9     return tf.stack(tf.shade(agg, cmap=['#333333', 'white'], how='eq_hist', alpha=255),
---> 10                           hillshade(agg, how='mdow', alpha=45))
     11 
     12 p = base_plot(x_range=(-11020645, -10797986), y_range=(3503546, 3632767))

/Users/bcollins/anaconda/envs/training/lib/python3.5/site-packages/datashader/transfer_functions.py in stack(*imgs, **kwargs)
     57         return imgs[0]
     58     imgs = xr.align(*imgs, copy=False, join='outer')
---> 59     out = tz.reduce(tz.flip(op), [i.data for i in imgs])
     60     return Image(out, coords=imgs[0].coords, dims=imgs[0].dims)
     61 

/Users/bcollins/anaconda/envs/training/lib/python3.5/site-packages/toolz/functoolz.py in __call__(self, *args, **kwargs)
    279     def __call__(self, *args, **kwargs):
    280         try:
--> 281             return self._partial(*args, **kwargs)
    282         except TypeError as exc:
    283             if self._should_curry(args, kwargs, exc):

/Users/bcollins/anaconda/envs/training/lib/python3.5/site-packages/toolz/functoolz.py in flip(func, a, b)
    619     [1, 2, 3]
    620     """
--> 621     return func(b, a)
    622 
    623 

TypeError: ufunc 'over' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
```

**Comment by jbednar** [comment]
Are you able to tell which data is integer in this case, and why it's not casting from int to float instead?

**Comment by brendancol** [comment]
@jbednar my understanding is that this is a change in numpy which makes you explicitly allow unsafe casting.  example: `numpy.multiply(A, B, out=A, casting='unsafe')`

**Comment by jbednar** [comment]
What I can't tell is why it's downcasting rather than upcasting.  Is this a case where the result *should* be an integer, or should it be a float?

**Comment by jbednar** [comment]
I was able to reproduce this bug only in the case of having a blank image.  Blank images use a different codepath that generates an Image with pixels of type float64, rather than uint32 as it should have been.  Fixed in commit 9843c77, unless there is some other case where images of type float64 are created somewhere.
